src/k8s-engine/wsl.ts

```diff
@@ -1306,6 +1309,50 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
     await this.execCommand('/usr/sbin/update-ca-certificates');
   }
 
+  /**
+   * Run provisioning scripts; this is done after init is started.
+   */
+  protected async runProvisioningScripts() {
+    const provisioningPath = path.join(paths.config, 'provisioning');
+
+    await fs.promises.mkdir(provisioningPath, { recursive: true });
+    await Promise.all([
+      (async() => {
+        // Write out the readme file.
+        const ReadmePath = path.join(provisioningPath, 'README');
+
+        try {
+          await fs.promises.access(ReadmePath, fs.constants.F_OK);
+        } catch {
+          const contents = `
+            Any files named '*.start' in this directory will be executed
+            sequentially on Rancher Desktop startup, before the main services.
+            Files are processed in lexical order, and will delay startup until
+            they are complete. Similaryly, any files named '*.stop' will be
+            executed on shutdown, after the main services have exited.
+            `.replace(/\s*\n\s*/g, '\n').trim();
+
+          await fs.promises.writeFile(ReadmePath, contents, { encoding: 'utf-8' });
+        }
+      })(),
+      (async() => {
+        const linuxPath = await this.wslify(provisioningPath);
+
+        // Run the provisioning job. We need to clobber /etc/local.d as
+        // /etc/init.d/local has no options to use a different directory.
+        await this.execCommand('/usr/local/bin/wsl-service', '--ifstarted', 'local', 'stop');
+        await this.execCommand('/bin/rm', '-r', '-f', '/etc/local.d');
+        await this.execCommand('/bin/ln', '-s', '-f', '-T', linuxPath, '/etc/local.d');
+        await this.execCommand(
+          '/bin/sh',
+          '-c',
+          `for f in '${ linuxPath }'/*.start '${ linuxPath }'/*.stop; do [ -f "\${f}" ] && chmod a+x "\${f}"; done`);
+        // This should not be running (because we tore down init), but to be safe...
+        await this.execCommand('/usr/local/bin/wsl-service', 'local', 'start');
```

Comment:
So should there be an `--ifstarted` flag here?

Reply:
No, because it makes no sense to start `local` if it's already started (it makes sense on `wsl-service local stop`, though).